tests/rag_answers/custom_deepeval/metrics/test_factual_correctness.py

```diff
@@ -0,0 +1,415 @@
+                (False, 0.7, ["fact1", "fact2"], [], 1.0, True),
+                (
+                    True,
+                    0.7,
+                    ["fact1", "fact2", "fact3"],
+                    ["fact4"],
+                    0.0,
+                    False,
+                ),  # strict: score < strict-threshold of 1
+                (
+                    True,
+                    0.7,
+                    ["fact1", "fact2"],
+                    [],
+                    1.0,
+                    True,
+                ),  # strict: score == strict-threshold of 1
+                (
+                    True,
+                    0.7,
+                    ["fact1"],
+                    ["fact2", "fact3"],
+                    0.0,
+                    False,
+                ),  # strict: score <= threshold so should not trigger strict mode logic
+            ],
+        )
+        async def test_strict_mode_can_be_configured(
+            self,
+            strict_mode: bool,
+            threshold: float,
+            TP: list,
+            FP: list,
+            expected_score: float,
+            expected_success: bool,
+            mock_native_model: Mock,
+            test_case: LLMTestCase,
+        ):
+            classified_facts = ClassifiedFacts(TP=TP, FP=FP, FN=[])
+
+            metric = FactualCorrectnessMetric(
+                model=mock_native_model,
+                threshold=threshold,
+                strict_mode=strict_mode,
+            )
+
+            metric._a_classify_statements = AsyncMock(return_value=classified_facts)
+
+            # let _calculate_score and _finalise_evaluation run naturally
+            result = await metric.a_measure(test_case)
+
+            assert round(result, 4) == round(expected_score, 4)
+            assert metric.success == (expected_success)
+
+        @pytest.mark.asyncio
+        async def test_native_model_sets_evaluation_costs(
+            self,
+            mock_native_model: Mock,
+            test_case: LLMTestCase,
+            fact_classification_result: FactClassificationResult,
+        ):
+            expected_cost = 0.05
+
+            mock_native_model.a_generate = AsyncMock(
+                return_value=(fact_classification_result, expected_cost)
+            )
+
+            metric = FactualCorrectnessMetric(model=mock_native_model)
+            metric.using_native_model = True  # ensure we hit native logic
+
+            _ = await metric.a_measure(test_case)
+
+            assert metric.evaluation_cost == expected_cost
+            mock_native_model.a_generate.assert_awaited_once()
+
+        @pytest.mark.asyncio
+        @patch(
+            "govuk_chat_evaluation.rag_answers.custom_deepeval.metrics.factual_correctness.factual_correctness.initialize_model"
+        )
+        @patch.object(
+            FactualCorrectnessTemplate, "classify_facts", return_value="mocked_prompt"
+        )
+        async def test_non_native_model_returns_valid_schema(
```

Comment:
For this test I think you'd be best to mock the a_generate function on the model to return the FactClassificationResult object